### PR TITLE
Add an annotation identifying upcoming nodes

### PR DESCRIPTION
**Pull request by towca**

### CODE REVIEW

File: cluster-autoscaler/core/static_autoscaler.go [hunk 1]
```diff
@@ -57,6 +57,9 @@ const (
 	unschedulablePodWithGpuTimeBuffer = 30 * time.Second
 	// How long should Cluster Autoscaler wait for nodes to become ready after start.
 	nodesNotReadyAfterStartTimeout = 10 * time.Minute
+
+	// NodeUpcomingAnnotation is an annotation CA adds to nodes which are upcoming.
+	NodeUpcomingAnnotation = "cloud.google.com/cluster-autoscaler-node-upcoming"
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/static_autoscaler.go:62]: This is in no way specific to GCP. Please use a more generic CA prefix.

Comment (towca) [reply cluster-autoscaler/core/static_autoscaler.go]: My bad, changed!

**Pull request merged**